libcxx/include/__expected/expected.h

```diff
@@ -909,6 +878,14 @@ class expected {
         std::__expected_construct_unexpected_from_invoke_tag, _Func&& __f, _Args&&... __args)
         : __unex_(std::invoke(std::forward<_Func>(__f), std::forward<_Args>(__args)...)) {}
 
+    template <class _Union>
+    _LIBCPP_HIDE_FROM_ABI constexpr explicit __union_t(bool __has_val, _Union&& __other) {
+      if (__has_val)
+        std::construct_at(this, std::in_place, std::forward<_Union>(__other).__val_);
```

Comment:
Wouldn't it be cleaner to use this instead?

```
if (__has_val)
  std::construct_at(&__val_, std::in_place, std::forward<_Union>(__other).__val_);
else
  std::construct_at(&__unex_, std::unexpect, std::forward<_Union>(__other).__unex_);
```

Reply:
Yes, that looks better -- delegating to the other constructors like this is probably a bit too clever.

Reply:
done!